kedro/config/omegaconf_config.py

```diff
@@ -217,6 +218,7 @@ def load_and_merge_dir_config(
         Args:
             conf_path: Path to configuration directory.
             patterns: List of glob patterns to match the filenames against.
+            key: Str of the dictionary keys to access.
```

Comment:
This is quite a generic description. Maybe change it to: "Key of the configuration type to fetch."